qiskit_nature/operators/second_quantization/quadratic_hamiltonian.py

```diff
@@ -0,0 +1,349 @@
+                and antisymmetric_part is not None
+                and hermitian_part.shape != antisymmetric_part.shape
+            ):
+                raise ValueError("Hermitian part and antisymmetric part must have same shape.")
+            if hermitian_part is not None:
+                if hermitian_part.shape[0] != self._num_modes:
+                    raise ValueError("Hermitian part must have shape num_modes x num_modes.")
+                if not _is_hermitian(hermitian_part, rtol=rtol, atol=atol):
+                    raise ValueError("Hermitian part must be Hermitian.")
+            if antisymmetric_part is not None:
+                if antisymmetric_part.shape[0] != self._num_modes:
+                    raise ValueError("Antisymmetric part must have shape num_modes x num_modes.")
+                if not _is_antisymmetric(antisymmetric_part, rtol=rtol, atol=atol):
+                    raise ValueError("Antisymmetric part must be antisymmetric.")
+
+        self._hermitian_part = hermitian_part
+        self._antisymmetric_part = antisymmetric_part
+        self._constant = constant
+
+        if self._hermitian_part is None:
+            self._hermitian_part = np.zeros((self._num_modes, self._num_modes))
+        if self._antisymmetric_part is None:
+            self._antisymmetric_part = np.zeros((self._num_modes, self._num_modes))
+
+    @property
+    def hermitian_part(self) -> np.ndarray:
+        """The matrix of coefficients of terms that conserve particle number."""
+        return self._hermitian_part
+
+    @property
+    def antisymmetric_part(self) -> np.ndarray:
+        """The matrix of coefficients of terms that do not conserve particle number."""
+        return self._antisymmetric_part
+
+    @property
+    def constant(self) -> float:
+        """The constant."""
+        return self._constant
+
+    @property
+    def num_modes(self) -> float:
+        """The number of modes this operator acts on."""
+        return self._num_modes
+
+    def to_fermionic_op(self) -> FermionicOp:
+        """Convert to FermionicOp."""
+        terms = [([], self.constant)]  # type: list[tuple[list[tuple[str, int]], complex]]
+        for i in range(self._num_modes):
+            terms.append(([("+", i), ("-", i)], self.hermitian_part[i, i]))
+            for j in range(i + 1, self._num_modes):
+                terms.append(([("+", i), ("-", j)], self.hermitian_part[i, j]))
+                terms.append(([("+", j), ("-", i)], self.hermitian_part[j, i]))
+                terms.append(([("+", i), ("+", j)], self.antisymmetric_part[i, j]))
+                terms.append(([("-", j), ("-", i)], self.antisymmetric_part[i, j].conjugate()))
+        return FermionicOp(terms, register_length=self._num_modes)
+
+    def conserves_particle_number(self) -> bool:
+        """Whether the Hamiltonian conserves particle number."""
+        return np.allclose(self.antisymmetric_part, 0.0)
+
+    def majorana_form(self) -> tuple[np.ndarray, float]:
```

Comment:
In order to be able to use a type hint such as this one in Python versions <3.9 you have to add the following import:
```python
from __future__ import annotations
```
That is what is causing the CI to break.

Reply:
Done.